url.py

```diff
@@ -322,6 +322,10 @@ def unpunycode(self):
     ###########################################################################
     # Information about the domain
     ###########################################################################
+    def hostname(self):
```

Comment:
I'll work on some tests to prove that and then hopefully roll out a fix in short order.